drizzlepac/hapsequencer.py

```diff
@@ -334,6 +335,23 @@ def run_hla_processing(input_filename, result=None, debug=False):
         # instrument_programID_obsetID_manifest.txt (e.g.,wfc3_b46_06_manifest.txt)
         manifest_name = total_list[0].manifest_name
 
+        # Set up all pipeline parameter values that will be used later on in the run.
+        # First, build temp.list of all product objects
+        full_product_list = total_list.copy()
+        for total_item in total_list:
+            full_product_list += total_item.fdp_list
+            full_product_list += total_item.edp_list
+
+        # Next, generate output parameter file
+        param_out_filename = input_filename.replace(".out", "_config.json")
+
+        # Finally, instantiate configuration object in each product object
+        for item in full_product_list:
+            item.pars = config_utils.HapConfig(item, output_custom_pars_file=param_out_filename, use_defaults=True)
```

Comment:
The HAPProduct definition in hlautils/product.py should be updated to change "self.param_dict={}" to "self.pars=None".  This will document that the pars object will be attached to the product once it gets instantiated by this line, rather than magically adding an attribute out of nowhere (which Python happily allows a developer to do much to the confusion of anyone trying to maintain that code later...).

Reply:
Resolved by commit [0a74d37](https://github.com/spacetelescope/drizzlepac/pull/439/commits/0a74d37a7d37f5e0bb7a86f373d69ca0bbd9bca9)